### ISSUE TITLE
Closures

### ISSUE THREAD

**Issue by RealyUniqueName** [RealyUniqueName/haxe#19]
Haxe

``` haxe
var notUsedInClosure = 123;
var used = "world";

return function () {
    this.hello(used);
}
```

PHP

``` php
$notUsedInClosure = 123;
$used = "world";

return function () use (&$used) {
    $this->hello($used);
};
```

[Anonymous functions](http://php.net/manual/en/functions.anonymous.php)
Support of `this` in closures requires at least PHP 5.4.

_Implemented in adff433037995917bebae91e7c0ff292959c092b and e9f152e878d127f347390196e3b676ad2371cb93_

**Comment by mockey** [comment]
This is one of the biggest weaknesses of the old genphp.. Except for passing local vars this should be straightforward, no?
PHP 5.4 is no problem, I'd say. 5.5 is the oldes supported version ATM.

**Comment by RealyUniqueName** [comment]
Yes, the only problem is collecting vars available in current scope.

**Comment by mockey** [comment]
There should be code for this in old genphp (as you probably know). The old closure generation needs this as well.

**Comment by RealyUniqueName** [comment]
Implemented in adff433 and e9f152e
